lib/messageformat.js

```diff
@@ -133,6 +133,36 @@ MessageFormat.formatters = {
   }
 };
 
+/** Predefined number formatting function enforcing Bidi Structured Text by using UCC
+ * Prevents intermingling of text contained in placeholders with the rest of message text.
+ * Maintains the integrity of Bidi text flowing according to directionality of language script.
+ * For reference to Bidi structured text see: http://cldr.unicode.org/development/development-process/design-proposals/bidi-handling-of-structured-text
+ *  @param {string} - Placeholder text content
+ *  @param {string} - Locale
+ *  @returns {string} - The resultant text fixed by UCC Bidi strong marks
+ *  @examples: (upper case stands for Bidi (Arabic/Hebrew) characters, output is shown as displayed in browser page
+ *  > var MessageFormat = require('messageformat');
+ *  > var mf = new MessageFormat('en', null, {"bidiStructuredText": MessageFormat.formatters.bidiStructuredText} );
+ *  > var mfunc = mf.compile("{0} >> {1} >> {2}");
+ *  > mfunc([ "first_english_word", "SECOND_ARABIC_WORD", "THIRD_ARABIC_WORD" ])
+ * structured output:
+ *     "first_english_word >> SECOND_ARABIC_WORD  >> THIRD_ARABIC_WORD"
+ * distorted output when MessageFormat gets initialized with no structured text formatter:
+ *     "first_english_word >> THIRD_ARABIC_WORD  << SECOND_ARABIC_WORD"
+ *  */
+MessageFormat.formatters.bidiStructuredText = function( text, locale ) { //shensis mesage_format
```

Comment:
This method is still in `MessageFormat.formatters`, and there's still no rationalisation for that. Also, there's a spurious comment at the end of this line.